pumpkin-inventory/src/player.rs

```diff
@@ -57,13 +57,19 @@ impl PlayerInventory {
             if !(0..=45).contains(&slot) {
                 Err(InventoryError::InvalidSlot)?
             }
-            *self.all_slots()[slot] = item;
+            let Some(slot) = self.get_slot_mut(slot) else {
```

Comment:
change this to `let slot = self.get_slot_mut(slot).ok_or(Err(InventoryError::InvalidSlot))?;`